cmd/jujud/agent/machine.go

```diff
@@ -225,14 +225,24 @@ func (a *machineAgentCmd) Init(args []string) error {
 		return errors.Annotate(err, "cannot read agent configuration")
 	}
 
+	config := a.currentConfig.CurrentConfig()
 	// the context's stderr is set as the loggo writer in github.com/juju/cmd/logging.go
 	a.ctx.Stderr = &lumberjack.Logger{
-		Filename:   agent.LogFilename(a.currentConfig.CurrentConfig()),
+		Filename:   agent.LogFilename(config),
 		MaxSize:    300, // megabytes
 		MaxBackups: 2,
 		Compress:   true,
 	}
 
+	if loggingOverride := config.Value(agent.LoggingOverride); loggingOverride != "" {
```

Comment:
There's a Validate() function for agent config - can we add validation there that any supplied config parses etc